x/stakeibc/types/message_toggle_trade_controller.go

```diff
@@ -0,0 +1,64 @@
+package types
+
+import (
+	"errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	errorsmod "cosmossdk.io/errors"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/Stride-Labs/stride/v19/utils"
+)
+
+const TypeMsgToggleTradeController = "toggle_trade_controller"
+
+var _ sdk.Msg = &MsgToggleTradeController{}
+
+func NewMsgToggleTradeController(creator, chainId string, permissionChange AuthzPermissionChange, address string) *MsgToggleTradeController {
+	return &MsgToggleTradeController{
+		Creator:          creator,
+		ChainId:          chainId,
+		PermissionChange: permissionChange,
+		Address:          address,
+	}
+}
+
+func (msg *MsgToggleTradeController) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgToggleTradeController) Type() string {
+	return TypeMsgToggleTradeController
+}
+
+func (msg *MsgToggleTradeController) GetSigners() []sdk.AccAddress {
+	creator, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{creator}
+}
+
+func (msg *MsgToggleTradeController) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgToggleTradeController) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		return errorsmod.Wrapf(sdkerrors.ErrInvalidAddress, "invalid creator address (%s)", err)
+	}
+	if err := utils.ValidateAdminAddress(msg.Creator); err != nil {
+		return err
+	}
+	if msg.ChainId == "" {
+		return errors.New("chain ID must be specified")
+	}
+	if msg.Address == "" {
```

Comment:
Should we also do a check with utils.ValidateAdminAddress on the Address passed in to be gaining permissions?  This would only allow us to use a limited but trusted set of addresses whereas right now we can run this many times and grant many addresses ability to trade even though it has to be an admin who is doing the granting.  Making the set limited to a chain upgrade might give more trust?

Reply:
hmm it's a good question. I think it'd be nice to have some flexibility here though. Hypothetically if the account was compromised, we wouldn't be able to update it to a new account until we did a chain upgrade

Reply:
Isn't the address passed in here supposed to be the account to which we're granting authz permissions on Osmosis? That would not pass a utils.ValidateAdminAddress check. Maybe you're proposing limiting Address to be in a separate list

Reply:
My understanding of the suggestion was that this would be a separate list of osmo addresses

Reply:
Makes sense. I lean toward maintaining some flexibility - admin is the main point of trust

Reply:
Agreed, it's a good suggestion but I also lean slightly towards letting the admin set any address as a trade address. Gives us a bit more flexibility in a nightmare scenario, and I think the "restricted trade list" only grants protection in a scenario where the admin is compromised but the trade routers aren't. This is relatively unlikely imo.

Reply:
I think that people who have weighed in are all in agreement just to use the admin to turn on or off operators as is without putting any additional constraints on the operator addresses themselves.  Resolving.